### ISSUE TITLE
[java] UselessStringValueOf: False positive when `valueOf(char [], int, int)` is used

### ISSUE THREAD

**Issue by pzygielo** [pmd/pmd#3491]
**Affects PMD Version:** 6.37.0

**Rule:** [UselessStringValueOf](https://pmd.github.io/latest/pmd_rules_java_performance.html#uselessstringvalueof)

**Description:** False positive for UselessStringValueOf when valueOf(char [], int, int) is used

**Code Sample demonstrating the issue:**
ab37b1ccb2d040eb782dead5f28679cbc9532499/https://github.com/pmd/pmd/pull/3490
```java
public class Test {
    public String test(char buf[], int offset, int len) {
        print("XML Parser characters: " + String.valueOf(buf, offset, len));
    }
}
```

**Expected outcome:**

PMD reports a violation at line 3, but that's wrong. That's a false positive.

**Running PMD through:** *Maven*

**Comment by pzygielo** [comment]
Thank you for the fix and new release.
